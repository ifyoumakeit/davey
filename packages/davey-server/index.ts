export { default as renderToStaticMarkup } from "./src/renderToStaticMarkup";
