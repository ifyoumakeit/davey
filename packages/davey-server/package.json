{
  "name": "davey-server",
  "version": "0.3.0",
  "description": "Dave's Alternate View Interpretation DOM helpers",
  "main": "./lib/index.js",
  "source": "./index.ts",
  "repository": "git@github.com:ifyoumakeit/davey.git",
  "author": "Dave Garwacke <dave.garwacke@warbyparker.com>",
  "license": "MIT"
}
