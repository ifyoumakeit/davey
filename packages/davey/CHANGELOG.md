# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.6.3"></a>
## <small>0.6.3 (2018-06-04)</small>

* docs: update package descriptions ([ce8e7b2](https://github.com/ifyoumakeit/davey/commit/ce8e7b2))




<a name="0.6.2"></a>
## <small>0.6.2 (2018-05-28)</small>

* ci(update ci): ([174a8cc](https://github.com/ifyoumakeit/davey/commit/174a8cc))
* revert(re-add period): ([767d041](https://github.com/ifyoumakeit/davey/commit/767d041))
* revert(re-try all of this madness): ([758bb05](https://github.com/ifyoumakeit/davey/commit/758bb05))
* revert(remove comment): ([a6d943d](https://github.com/ifyoumakeit/davey/commit/a6d943d))
* refactor: chill ([5e0f7d2](https://github.com/ifyoumakeit/davey/commit/5e0f7d2))
* revert: re-add comment ([be5b78d](https://github.com/ifyoumakeit/davey/commit/be5b78d))
* chore: remove comment ([79bee27](https://github.com/ifyoumakeit/davey/commit/79bee27))
* ci: try to re-run ci ([fd39c27](https://github.com/ifyoumakeit/davey/commit/fd39c27))
* style: remove period ([e622df4](https://github.com/ifyoumakeit/davey/commit/e622df4))




<a name="0.5.0"></a>
# [0.5.0](https://github.com/ifyoumakeit/davey/compare/v0.4.5...v0.5.0) (2018-05-27)


### Features

* release via lerna ([eab4aa2](https://github.com/ifyoumakeit/davey/commit/eab4aa2))


### BREAKING CHANGES

* This will update to 1.0.0
