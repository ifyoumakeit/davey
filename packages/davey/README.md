# 👦🏻 davey

**Dave's alternate view engine yeah!**

Using my brain to recreate React, Redux and CSS-in-JS from scratch/memory. I promise not to read any React or Redux docs or articles while creating this. This is the analog to `React`.

## JSX

### Use w/ Typscript

1.  Add below your **tsconfig.json**

```json
{
  "compilerOptions": {
    "jsx": "react",
    "jsxFactory": "davey"
  }
}
```

### Use w/ Babel

1.  `yarn add --dev transform-react-jsx`
2.  Add below your **.babelrc**

```json
{
  "plugins": [["transform-react-jsx", { "pragma": "davey" }]]
}
```
