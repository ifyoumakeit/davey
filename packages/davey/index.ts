export interface Props {
  children?: string[];
}

export type Tag = Function | string;

export const davey = (tag: Tag, _props: Props = {}, ...rest: string[]) => {
  // Get children from third argument or from props
  const _children = rest.length ? rest : (_props && _props.children) || [];

  const props = {
    ..._props,
    children: []
      .concat(_children)
      .reduce((acc, child) => acc.concat(child), []),
  };

  return typeof tag === "function" ? tag(props) : { tag, props };
};
