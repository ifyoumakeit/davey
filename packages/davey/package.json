{
  "name": "davey",
  "version": "0.2.2",
  "description": "Dave's Alternate View Interpretation",
  "main": "./lib/index.js",
  "source": "./index.ts",
  "repository": "git@github.com:ifyoumakeit/davey.git",
  "author": "Dave Garwacke <dave.garwacke@warbyparker.com>",
  "license": "MIT"
}
