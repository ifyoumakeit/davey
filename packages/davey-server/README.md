# 👦🏻 davey-server

**davey's server methods**

## Methods

### renderToStaticMarkup
Like React's `renderToStaticMarkup`, takes a davey component and returns the resolved markup.

## Try it out

`yarn add davey davey-server`

```
import { davey } from "davey";
import { renderToStaticMarkup } from "davey-server";

renderToStaticMarkup(<h1>Hello world!</h1>);
```
