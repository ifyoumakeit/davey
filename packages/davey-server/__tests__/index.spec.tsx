import { davey } from "davey";
import { renderToStaticMarkup } from "davey-server";

describe("Basic single element", () => {
  test("match snapshot", () => {
    expect(renderToStaticMarkup(<h1>hello</h1>)).toMatchSnapshot();
  });
});

describe("2-level deep nested element", () => {
  test("match snapshot", () => {
    expect(
      renderToStaticMarkup(
        <main>
          <header>
            <h1>hello</h1>
            <h2>world</h2>
          </header>
        </main>
      )
    ).toMatchSnapshot();
  });
});

describe("Resolves text nodes ", () => {
  test("match snapshot", () => {
    expect(
      renderToStaticMarkup(
        <h1>
          hello <span>my</span>
          <h2>world</h2> friend
        </h1>
      )
    ).toMatchSnapshot();
  });
});

describe("Appends attributes", () => {
  let markup;

  beforeAll(() => {
    markup = renderToStaticMarkup(
      <button
        onClick={() => expect(true).toEqual(true)}
        onBlur={() => expect(true).toEqual(true)}
        style={{ backgroundColor: "blue" }}
      >
        hello
      </button>
    );
    document.body.innerHTML = markup;
  });
  test("match snapshot", () => {
    expect(markup).toMatchSnapshot();
  });

  test("onBlur isn't present", () => {
    expect(
      document.body.querySelector("button").getAttribute("onClick")
    ).toBeFalsy();
  });

  test("style works", () => {
    expect(
      document.body.querySelector("button").style["background-color"]
    ).toBe("blue");
  });
});
