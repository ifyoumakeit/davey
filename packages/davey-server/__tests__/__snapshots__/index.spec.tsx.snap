// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2-level deep nested element match snapshot 1`] = `
"<main>
	<header>
		<h1>
			hello
		</h1>
		<h2>
			world
		</h2>
	</header>
</main>"
`;

exports[`Basic single element match snapshot 1`] = `
"<h1>
	hello
</h1>"
`;

exports[`Resolves text nodes  match snapshot 1`] = `
"<h1>
	hello 
	<span>
		my
	</span>
	<h2>
		world
	</h2>
	friend
</h1>"
`;
